### PR TITLE
EVG-2141 add system-out and system-err to xunit logs without panickin…

### PR DESCRIPTION
**Pull request by john-m-liu**
…g agent

### CODE REVIEW

File: command/results_xunit.go [hunk 1]
```diff
@@ -130,35 +129,60 @@ func (c *xunitResults) parseAndUploadResults(ctx context.Context, conf *model.Ta
...
+		for _, tc := range suite.TestCases {
+			// logs are only created when a test case does not succeed
+			test, log := tc.toModelTestResultAndLog(t)
+			tests = append(tests, test)
+			if log != nil {
+				if suite.SysOut != "" {
+					log.Lines = append(log.Lines, fmt.Sprintf("system-out: %s", suite.SysOut))
+				}
+				if suite.SysErr != "" {
+					log.Lines = append(log.Lines, fmt.Sprintf("system-err: %s", suite.SysErr))
```
Comment (tychoish) [inline command/results_xunit.go]: append(log.Lines, "system-err:", suite.SysErr)

File: command/results_xunit.go [hunk 2]
```diff
@@ -130,35 +129,60 @@ func (c *xunitResults) parseAndUploadResults(ctx context.Context, conf *model.Ta
...
+				}
+				logs = append(logs, log)
+				logIdxToTestIdx[len(logs)-1] = len(tests) - 1
```
Comment (tychoish) [inline command/results_xunit.go]: don't we want the map in the other direction? multiple tests could have logs of the same length?

File: command/results_xunit.go (hunk 1 above)
Comment (john-m-liu) [reply command/results_xunit.go]: fixed

File: command/results_xunit.go (hunk 2 above)
Comment (john-m-liu) [reply command/results_xunit.go]: With the data structure used here, there is a 1:1 mapping of test struct to log struct (see line 172). A log:test mapping is more convenient for how it's used here

**Comment by tychoish** [comment]
the mongo-release-tools project is in staging, which uses xunit, and might be a good thing to test.

**Pull request merged**